app/models/dialog/container_template_service_dialog.rb

```diff
@@ -0,0 +1,47 @@
+class Dialog
+  class ContainerTemplateServiceDialog
+    def self.create_dialog(label, params)
+      new.create_dialog(label, params)
+    end
+
+    # This dialog is to be used by a container template service
+    def create_dialog(label, params)
+      Dialog.new(:label => label, :buttons => "submit,cancel").tap do |dialog|
+        tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+        add_parameters_group(tab, 0, params)
+        dialog.save!
+      end
+    end
+
+    private
+
+    def add_parameters_group(tab, position, params)
+      tab.dialog_groups.build(
+        :display  => "edit",
+        :label    => "Parameters",
+        :position => position
+      ).tap do |dialog_group|
+        params.each_with_index do |param, index|
+          add_parameter_field(param, dialog_group, index)
+        end
+      end
+    end
+
+    def add_parameter_field(param, group, position)
+      group.dialog_fields.build(
```

Comment:
The regex in ```from``` is used by Openshift to generate the value if empty. It is not used for validation. Take the above ```MQ_PASSWORD``` as an example which has ```from: "[a-zA-Z0-9]{8}"```. If the user leave the field empty, the value would be generated, like ```K0uVE3wN```. If the user input the value like ```abc```, or ```abcde12345``` in the template, they would all be accepted as valid by Openshift.